[SHK-31] 사용자는 자신의 대기열 순위를 확인하고 싶다. (들어갔을 때 순위)

goseungwon opened the pull request:
### ⛏ 작업 사항
- Sorted Set add 연산시 Sorted Set Size를 반환한다.
- Coupon 발급시 amount 1 감소
- Coupon 수량 상태 check

### 📝 작업 요약
- Coupon 관련 기능 추가

### 💡 관련 이슈
- Resolved : [SHK-42], [SHK-181]


[SHK-42]: https://shoekream.atlassian.net/browse/SHK-42?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SHK-181]: https://shoekream.atlassian.net/browse/SHK-181?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

goseungwon committed:
Feat : [SHK-42] 이벤트 요청시 사용자가 요청 순서를 리턴한다.
- 쿠폰 발급시 amount 가 0 초과하면 1 감소
- amount 가 0 이하인치 확인하는 메서드 생성

goseungwon committed:
Test : [SHK-181] 이벤트 요청시 사용자가 요청 순서를 리턴 테스트
- 쿠폰 개수 테스트 및 상태 테스트

goseungwon committed:
Merge remote-tracking branch 'origin/feature/SHK-31' into feature/SHK-31

goseungwon commented:
안녕하세요 여러분,, 커밋 메세지랑 내용이랑 조금 다르죠..
커밋 단위가 꼬여버려서 어쩔 수 없는 선택을 했습니다 😭

boompatron commented on src/main/java/com/prgrms/kream/domain/coupon/model/Coupon.java:
```diff
@@ -43,11 +43,14 @@ public Coupon(Long id, int discountValue, String name, int amount) {
 	}
 
 	public void decreaseAmount() {
```
감소하는 로직을 변경하셨군요!
직접 값에 접근하는 것 보다는 플래그를 만드는게 더 효율적인가요???

goseungwon replied:
매번 체크하는 방법보다 플래그를 통해 체크를 안하는게 효율적이라 생각합니다!

boompatron reviewed (approved):
구현하느라 수고하셨습니다!! 질문 1개 남겼습니다!!

BnDC reviewed (approved):
구현하느라 고생하셨습니다. 딱히 코멘트할 것이 없네요

goseungwon committed:
Refactor : properties, CouponEventRegisterRequest 수정
- properties setter lombok이 아닌 직접 지정
- CouponEventRegisterRequest 필드 Long type으로 변경
